### ISSUE TITLE
RuntimeError: Cannot get window extent w/o renderer

### ISSUE THREAD

**Issue by tundo91** [mwaskom/seaborn#545]
I'm trying to plot a *seaborn* *clustermap* (it doesn't work with the *heatmap* too) with the following, no NaNs admitted:

```python
    import numpy as np
    import pandas as pd
    import seaborn as sns
    import matplotlib.pyplot as plt
    
    def plotClusterMap():
        a = pd.DataFrame(np.matrix('1 2; 3 4'))
        print a
        fig = plt.figure()
        sns.clustermap(a)
        plt.show()
```

```a``` is well formed:

       0  1
    0  1  2
    1  3  4

Console output:

    Traceback (most recent call last):
      File "main.py", line 78, in <module>
        main()
      File "main.py", line 72, in main
        heatmapPlotter.plotClusterMap()
      File "/Users/username/code.py", line 12, in plotClusterMap
        sns.clustermap(a)
      File "/Library/Python/2.7/site-packages/seaborn/matrix.py", line 895, in clustermap
        **kwargs)
      File "/Library/Python/2.7/site-packages/seaborn/matrix.py", line 813, in plot
        self.plot_matrix(colorbar_kws, mask, **kws)
      File "/Library/Python/2.7/site-packages/seaborn/matrix.py", line 803, in plot_matrix
        cbar_kws=colorbar_kws, mask=mask, **kws)
      File "/Library/Python/2.7/site-packages/seaborn/matrix.py", line 292, in heatmap
        plotter.plot(ax, cbar_ax, kwargs)
      File "/Library/Python/2.7/site-packages/seaborn/matrix.py", line 177, in plot
        if axis_ticklabels_overlap(xtl):
      File "/Library/Python/2.7/site-packages/seaborn/utils.py", line 374, in axis_ticklabels_overlap
        bboxes = [l.get_window_extent() for l in labels]
      File "/usr/local/lib/python2.7/site-packages/matplotlib/text.py", line 796, in get_window_extent
        raise RuntimeError('Cannot get window extent w/o renderer')
    RuntimeError: Cannot get window extent w/o renderer

* Mac OSX: 10.10.3
* Seaborn: 0.5.1
* Matplotlib: 1.3.1 (just downgraded from 1.4)

**Comment by tundo91** [comment]
Python 3.4.2 and numpy 1.9.2 don't solve the error :(

**Comment by tundo91** [comment]
[Workaround here...](http://stackoverflow.com/questions/30138026/seaborn-matplotlib-cannot-get-window-extent-w-o-renderer-runtimeerror)

**Comment by mwaskom** [comment]
Yeah as far as I understand this is a known issue in the matplotlib MacOSX backend.

**Comment by eyadsibai** [comment]
actually u could do matplotlib.use('agg')
This is a bug in matplotlib in Mac OS X

**Comment by mwaskom** [comment]
Closed with #611
